### PR TITLE
Fix some man page typos

### PR DESCRIPTION
**Pull request by mattcaswell**
A few miscellaneous man page typos reported by Hal Murray on openssl-users.

### CODE REVIEW

File: doc/man3/EVP_MAC.pod [hunk 1]
```diff
@@ -317,12 +317,14 @@ EVP_MAC_do_all_provided() returns nothing at all.
...
       if (cipher != NULL)
           params[params_n++] =
-              OSSL_PARAM_construct_utf8_string("cipher", cipher, 0, NULL);
+              OSSL_PARAM_construct_utf8_string("cipher", cipher,
+                                               strlen(cipher) + 1);
       if (digest != NULL)
           params[params_n++] =
-              OSSL_PARAM_construct_utf8_string("digest", digest, 0, NULL);
+              OSSL_PARAM_construct_utf8_string("digest", digest,
+                                               strlen(digest) + 1);
```
Comment (levitte) [inline doc/man3/EVP_MAC.pod:325]: You could actually make that 0, since `OSSL_PARAM_construct_utf8_string` will calculate the size itself in that case

Comment (paulidale) [reply doc/man3/EVP_MAC.pod:325]: Definitely do this, always pass 0 instead of calling strlen.

**Comment by mattcaswell** [comment]
Fixup pushed addressing feedback. @beldmit please can you reconfirm?

**Comment by mattcaswell** [comment]
@beldmit please can you reconfirm?....or approval anyone?

**Comment by beldmit** [comment]
Squashed and merged. Thanks!

**Pull request closed without merging**